### PR TITLE
Switch to nested object pluralisation format for i18n files

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/element-web/issues/25608
Needs https://github.com/vector-im/element-desktop/pull/1172
Needs https://github.com/vector-im/element-web/pull/25970
~Needs https://github.com/matrix-org/matrix-web-i18n/pull/7~
~Needs https://github.com/matrix-org/matrix-react-sdk/pull/11417~

Performed using `find src/i18n/strings -type f -exec sh -c 'jq \'to_entries | map(.key |= split("|")) | reduce .[] as $obj({ }; setpath($obj.key; $obj.value))\' {} | sponge {}' \;`

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/setup/setupLanguage.ts [hunk 1]
```diff
@@ -30,9 +30,10 @@ const lv = {
 // de_DE.json
 // lv.json - mock version with few translations, used to test fallback translation
 
-type Translations = Record<string, Record<string, string> | string>;
+type Translation = Partial<Record<"one" | "other", string>> | string;
```
Comment (richvdh) [inline test/setup/setupLanguage.ts:33]: I see "zero" in some places

**Comment by richvdh** [comment]
what's the story with the removed perl scripts? Are they no longer used? since when?

**Review by richvdh — COMMENTED** [review]
Help me out here, as a reviewer unfamiliar with this area: does the new format Just Work, or do we have to upgrade some other component to make it understand?

Indeed, is the new format documented anywhere?

**Comment by t3chguy** [comment]
> does the new format Just Work

Yup, counterpart uses this format by default, we also configure counterpart to treat `|` as a delimiter so they're functionally equivalent for counterpart. We don't remove that delimiter as the default is `.` and our keys are English sentences which contain `.`s so `.` as a delimiter would break things

**Comment by t3chguy** [comment]
> what's the story with the removed perl scripts? Are they no longer used? since when?

fix-i18n.pl I'm fairly sure was a one time thing
fixup-imports.pl I've never seen used

File: test/setup/setupLanguage.ts (hunk 1 above)
Comment (t3chguy) [reply test/setup/setupLanguage.ts:33]: Turns out its easier to handle these strings before they become nested from a find-and-replace perspective, so https://github.com/matrix-org/matrix-react-sdk/pull/11417 will handle that and then will update this PR

**Comment by t3chguy** [comment]
image

Confirmed Weblate is happy with this adjusted format, it uses dot notation for keys

**Pull request merged**